Sources/Private/CoreAnimation/Animations/CombinedShapeAnimation.swift

```diff
@@ -51,3 +51,25 @@ final class CombinedShapeItem: ShapeItem {
   let shapes: KeyframeGroup<[BezierPath]>
 
 }
+
+extension CombinedShapeItem {
+  /// Manually combines the given shape keyframes by manually interpolating at each frame
+  static func manuallyInterpolating(
+    shapes: [KeyframeGroup<BezierPath>],
+    name: String,
+    context: LayerContext)
+    -> CombinedShapeItem
+  {
+    let interpolators = shapes.map { KeyframeInterpolator<BezierPath>(keyframes: $0.keyframes) }
```

Comment:
Nice use of some infra from the main thread renderer! Wondering if we should move it to a shared place now

Reply:
Good idea, I'll move this to `Sources/Private/Utility/Interpolatable`